app/services/user_books/query_service.rb

```diff
@@ -7,15 +7,33 @@ def initialize(params)
     end
 
     def search
-      if @finish_date
-        return UserBook.finished(@user)
-      else
-        return UserBook.to_read(@user)
+      list =  @finish_date ? UserBook.finished(@user) : UserBook.to_read(@user)
+      new_list = []
+      id_array = list.pluck(:id)
+      rating_array = list.pluck(:rating)
+      notes_array = list.pluck(:notes)
+
+      options = { include: [:book] }
+      books = serialize_data(list)
+
+      books.each_with_index do |book, i|
+        new_book = book.dup
+        new_book[:user_book_id] = id_array[i]
```

Comment:
```
attribute :name_with_year do |object|
    "#{object.name} (#{object.year})"
  end
```
I got the sample code from `fast_jsonapi`
What I can do with this is to use `delegate`

```
# model file

delegate :id, :rating, :notes, to: :user_books, prefix: :user_book # can do this

```

```
attribute :name_with_year do |object|
    object.user_book_id
end
```

Reply:
One of the solutions I would propose is to use `Blueprinter` because it has a better way of showing the data

{
  books: {
   name: 'book1',
   author: {
    name: 'author'
   }
}

Reply:
nice! I'll try this one. will let you know how it turns out